speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/FileDialog.hpp

```diff
@@ -0,0 +1,269 @@
+#pragma once
+
+// includes that are necessary for windows
+#include <shobjidl.h>  // For Windows COM interface
+#include <locale>
+#include <codecvt>
+
+#define MAX_PATH_SIZE 260
+
+/// @brief              Releases a file dialog, deletes the filterSpec array and uninitializes the COM interface
+/// @param p_fileDialog The dialog to release
+/// @param p_filterSpec The filterSpec array to delete
+inline void Release(IFileDialog* p_fileDialog, COMDLG_FILTERSPEC* p_filterSpec)
+{
+    delete[] p_filterSpec;
+    p_fileDialog->Release();
+    CoUninitialize();
+}
+
+/// @brief              Releases a shell item and call Release with the remaining parameters
+/// @param p_shellItem  The shell item to release
+/// @param p_fileDialog The dialog to release
+/// @param p_filterSpec The filterSpec array to delete
+inline void Release(IShellItem* p_shellItem, IFileDialog* p_fileDialog, COMDLG_FILTERSPEC* p_filterSpec)
+{
+    p_shellItem->Release();
+    Release(p_fileDialog, p_filterSpec);
+}
+
+/// @brief            Opens a file dialog for the user to select a file, limiting the files shown to the provided parameters
+/// @param p_buf      A buffer to write the filename to
+/// @param p_names    The names of the types of file to select
+/// @param p_exts     The extension filters of the types of file to select
+///	@param p_extCount The amount of extensions/names provided
+/// @param p_err      A buffer to write an error to (if applicable)
+/// @note             See SelectBlackBox in ResearcherMenu.cpp for an example on how to call this function
+inline bool SelectFile(char* p_buf, const wchar_t** p_names, const wchar_t** p_exts, int p_extCount, char* p_err)
```

Comment:
This function is very similar to the `SelectFile` below, the only difference is the amount of parameters. Maybe try looking into merging this into 1 function with default parameters for the additional ones (those would default to null for the p_names, p_exts and to 0 in p_extCount)

But i'll leave this up to you whether you have time to refactor this.